src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -71,99 +83,308 @@ public long BytesWritten
+            }
+            else
             {
-                _bytesWritten = BytesWritten,
-                _bytesCommitted = BytesCommitted,
-                _inObject = _inObject,
-                _isNotPrimitive = _isNotPrimitive,
-                _tokenType = _tokenType,
-                _currentDepth = _currentDepth,
-                _writerOptions = _writerOptions,
-                _bitStack = _bitStack,
-            };
+                _arrayBufferWriter.Clear();
+            }
+            _output = _arrayBufferWriter;
+
+            ResetHelper();
         }
 
         /// <summary>
-        /// Constructs a new <see cref="Utf8JsonWriter"/> instance with a specified <paramref name="bufferWriter"/>.
+        /// Resets the <see cref="Utf8JsonWriter"/> internal state so that it can be re-used with the new instance of <see cref="IBufferWriter{Byte}" />.
         /// </summary>
         /// <param name="bufferWriter">An instance of <see cref="IBufferWriter{Byte}" /> used as a destination for writing JSON text into.</param>
-        /// <param name="state">If this is the first call to the ctor, pass in a default state. Otherwise,
-        /// capture the state from the previous instance of the <see cref="Utf8JsonWriter"/> and pass that back.</param>
+        /// <remarks>
+        /// The <see cref="Utf8JsonWriter"/> will continue to use the original writer options
+        /// but now write to the passed in <see cref="IBufferWriter{Byte}" /> as the new destination.
+        /// </remarks>
         /// <exception cref="ArgumentNullException">
         /// Thrown when the instance of <see cref="IBufferWriter{Byte}" /> that is passed in is null.
         /// </exception>
-        /// <remarks>
-        /// Since this type is a ref struct, it is a stack-only type and all the limitations of ref structs apply to it.
-        /// This is the reason why the ctor accepts a <see cref="JsonWriterState"/>.
-        /// </remarks>
-        public Utf8JsonWriter(IBufferWriter<byte> bufferWriter, JsonWriterState state = default)
+        /// <exception cref="ObjectDisposedException">
+        ///   The instance of <see cref="Utf8JsonWriter"/> has been disposed.
+        /// </exception>
+        public void Reset(IBufferWriter<byte> bufferWriter)
         {
+            CheckNotDisposed();
+
             _output = bufferWriter ?? throw new ArgumentNullException(nameof(bufferWriter));
-            _buffered = 0;
-            BytesCommitted = 0;
-            _buffer = _output.GetSpan();
+            _stream = default;
+
+            if (_arrayBufferWriter != null)
+            {
+                _arrayBufferWriter.Clear();
+            }
 
-            _inObject = state._inObject;
-            _isNotPrimitive = state._isNotPrimitive;
-            _tokenType = state._tokenType;
-            _writerOptions = state._writerOptions;
-            _bitStack = state._bitStack;
+            _arrayBufferWriter = default;
```

Comment:
Why do we clear it if we're just going to drop it?

Reply:
And if we must clear it for some reason (we don't do this in most other situations where we just let the GC clean up), this line should be moved into the if block.  It should also use `null` instead of `default`.